doc/proposals/olm-cmd-workflows.md

```diff
@@ -0,0 +1,120 @@
+
+The proposal is aimed at expalining the resulting process of developing an olm enabled operator from scratch with the above mentioned improvements.
+
+## Proposal
+
+### README
+
+The following document discusses workflows for creating a new operator and running it with OLM:
+
+Operator creation workflow:
+
+1. Create a new operator project using the SDK Command Line Interface(CLI)
+2. Define new resource APIs by adding Custom Resource Definitions(CRD)
+3. Define Controllers to watch and reconcile resources
+4. Write the reconciling logic for your Controller using the SDK and controller-runtime APIs
+5. Use the SDK CLI to build and generate the operator deployment manifests
+
+OLM Workflow:
+
+1. Use `operator-sdk generate bundle` to generate kustomize base templates with CSV metadata, build and push the operator bundle image to the specified registry.
+  * If the kustomize template having the operator and UI metadata is present, it is utilized along with the existing CRDs to build the operator image bundle.
+  * If the kustomize template is not present, provide inputs regarding the UI metadata to the interactive prompts appearing further.
+2. Use `operator-sdk run --olm` to generate kustomize base template with operator metadata, build and run the operator.
+
+### Create and deploy an app-operator
+
+```sh
+# Create an app-operator project that defines the App CR.
+$ mkdir -p $HOME/projects/example-inc/
+# Create a new app-operator project
+$ cd $HOME/projects/example-inc/
+$ operator-sdk new app-operator --repo github.com/example-inc/app-operator
+$ cd app-operator
+
+# Add a new API for the custom resource AppService
+$ operator-sdk add api --api-version=app.example.com/v1alpha1 --kind=AppService
+
+# Add a new controller that watches for AppService
+$ operator-sdk add controller --api-version=app.example.com/v1alpha1 --kind=AppService
+
+# Set the username variable
+$ export USERNAME=<username>
+
+# Build and push the app-operator image to a public registry such as quay.io
+$ operator-sdk build quay.io/$USERNAME/app-operator
+
+# Login to public registry such as quay.io
+$ docker login quay.io
+
+# Push image
+$ docker push quay.io/$USERNAME/app-operator
+
+# Generate kustomize base templates with UI metadata and publish operator in a bundle image
+# If you would like to write the operator manifests on disk, specify the path using the flag "--output-dir" 
+$ operator-sdk generate bundle <IMAGE_REGISTRY>/<TAG>
+      --package <Name of the package that bundle image belongs to>
+      --kustomize-dir <Path to the directory containing kustomize base templates [default:./config/]>
+      --output-dir <Optional output directory if operator manifests are to be written on disk>
+
+# If the kustomize templates do not contain UI metadata for populating CSV, provide inputs for the interactive prompts that would appear further		
+$ Provide DisplayName for your operator:		
+$ Provide version for your operator:		
+$ Provide minKubeVersion for CSV:		
```

Comment:
If these are new prompts we are asking the user, are we thinking about internationalizing the strings? We should at least call out what our plan is for i18n.

https://github.com/leonelquinteros/gotext

Reply:
Will there be input validation? If so, how do we ask the user to re-enter?

Reply:
These are the UI metadata fields which could asked along with the command. An exhaustive list is here: https://docs.google.com/document/d/1dP6Y9DlE3TuoVXaOdYbuKjo97Qxwqy-07A7msf6_cTQ/edit
We haven't discussed about the implementation yet. Should it be included in the doc?